### PR TITLE
8-3-2020 task kristan

### PR DESCRIPTION
**Pull request by kristan-dev**
- Updated KafkaCloudListener.java with new KafkaListener for CustomerUpdate, implemented switch with 3 cases for 3 existing topics (CUST_optOut_optIn, CUST_update_email, PROD_offer)
- Updated KafkaConsumerConfig.java with ConcurrentKafkaListenerContainerFactory Bean for customerUpdateFactory

### CODE REVIEW

File: src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java [hunk 1]
```diff
@@ -15,6 +16,8 @@
     public static final String CUSTOMER_OPT = "customerOpt";
     public static final String UPDATE_EMAIL = "updateEmail";
     public static final String PROD_OFFER = "productOffer";
+    public static final String CUSTOMER_UPDATE = "customerUpdate";
```
Comment (jmcloud20) [inline src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java]: Remove customer update if not used.

File: src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java [hunk 2]
```diff
@@ -46,5 +49,24 @@ public void readProdffer(String message) throws JsonProcessingException {
         log.info("Product offer message: " + message);
     }
 
+    @KafkaListener(topics = "CustomerUpdate", groupId = "customerUpdate", containerFactory = "customerUpdateFactory")
+    public void customerUpdate(String message) throws JsonProcessingException {
+        MessageDto messageDto = messageService.createMessageDto(message);
 
+        String topic = messageDto.getTopic();
+        switch (topic) {
```
Comment (jmcloud20) [inline src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java]: Change to switch(messageDto.getTopic()){

File: src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java (hunk 1 above)
Comment (kristan-dev) [reply src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java]: removed

File: src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java (hunk 2 above)
Comment (kristan-dev) [reply src/main/java/com/pccw/cloud/consumerapp/KafkaCloudListener.java]: updated. thanks for feedback

**Review by jmcloud20 — APPROVED** [review]
No findings found.

**Pull request merged**